Everything works with and without resume except for SPN_1ADV which fails for both. I get this error 

```Traceback (most recent call last):
  File "scripts/train_agents.py", line 357, in <module>
    SPN_1ADV(args=args)
  File "scripts/train_agents.py", line 56, in SPN_1ADV
    attack_rounds=attack_rounds
  File "/home/naren/Developer/py/mhri/multiHRI/scripts/utils/train_helper.py", line 96, in get_N_X_SP_agents
    n_x_sp_eval_types=n_x_sp_eval_types
  File "/home/naren/Developer/py/mhri/multiHRI/scripts/utils/train_helper.py", line 146, in joint_ADV_N_X_SP
    use_entire_population_for_train_types_teammates=True)
  File "/home/naren/Developer/py/mhri/multiHRI/oai_agents/common/teammates_collection.py", line 240, in generate_TC
    use_entire_population=use_entire_population_for_train_types_teammates
  File "/home/naren/Developer/py/mhri/multiHRI/oai_agents/common/teammates_collection.py", line 41, in get_teammates
    assert len(agents_perftag_score) % required_population_size == 0, \
ZeroDivisionError: integer division or modulo by zero
```
(this is with quick test) 

Also I looked at resume for the other `RLAgentTrainer` instances. Seems like it could be done where if resume is enabled instead of returning the loaded agent you restart training, but I think that should be in a different PR

--- a/oai_agents/agents/base_agent.py
+++ b/oai_agents/agents/base_agent.py
@@ -511,7 +511,6 @@ class OAITrainer(ABC):
         # Load weights
         agents = []
         for agent_fn in saved_variables['agent_fns']:
-            print(f"Loading agent from: {agent_path / agent_fn}")
             agent = load_agent(agent_path / agent_fn, args)
             agent.to(device)
             agents.append(agent)

--- a/oai_agents/agents/rl.py
+++ b/oai_agents/agents/rl.py
@@ -12,6 +12,7 @@ from stable_baselines3.common.env_util import make_vec_env
 from stable_baselines3.common.vec_env import DummyVecEnv, SubprocVecEnv
 from sb3_contrib import RecurrentPPO, MaskablePPO
 import wandb
+import os
 
 VEC_ENV_CLS = DummyVecEnv #
 
@@ -54,8 +55,10 @@ class RLAgentTrainer(OAITrainer):
 
         self.learner_type = learner_type
         self.env, self.eval_envs = self.get_envs(env, eval_envs, deterministic, learner_type, start_timestep)
+        # Episode to start training from (usually 0 unless restarted)
         self.start_step = start_step
         self.steps = self.start_step
+        # Cumm. timestep to start training from (usually 0 unless restarted)
         self.start_timestep = start_timestep
 
         self.learning_agent, self.agents = self.get_learning_agent(agent)
@@ -289,7 +292,7 @@ class RLAgentTrainer(OAITrainer):
         print("Final sparse reward ratio: ", self.args.final_sparse_r_ratio)
 
 
-    def train_agents(self, total_train_timesteps, tag, exp_name=None, resume_ck_list=None):
+    def train_agents(self, total_train_timesteps, tag_for_returning_agent, exp_name=None, resume_ck_list=None):
         experiment_name = self.get_experiment_name(exp_name)
         run = wandb.init(project="overcooked_ai", entity=self.args.wandb_ent, dir=str(self.args.base_dir / 'wandb'),
                          reinit=True, name=experiment_name, mode=self.args.wandb_mode,
@@ -299,8 +302,6 @@ class RLAgentTrainer(OAITrainer):
 
         if self.checkpoint_rate is not None:
             if self.args.resume:
-                import os
-
                 path = self.args.base_dir / 'agent_models' / experiment_name
 
                 ckpts = [name for name in os.listdir(path) if name.startswith("ck")]
@@ -354,8 +355,8 @@ class RLAgentTrainer(OAITrainer):
                     print(f'New best evaluation score of {mean_reward} reached, model saved to {best_path}/{best_tag}')
                     self.best_score = mean_reward
             self.steps += 1
-        self.save_agents()
-        self.agents, _, _ = RLAgentTrainer.load_agents(args=self.args, name=self.name, tag=tag)
+        self.save_agents(tag=KeyCheckpoints.MOST_RECENT_TRAINED_MODEL)
+        self.agents, _, _ = RLAgentTrainer.load_agents(args=self.args, name=self.name, tag=tag_for_returning_agent)
         run.finish()
 
     @staticmethod

--- a/oai_agents/common/arguments.py
+++ b/oai_agents/common/arguments.py
@@ -35,7 +35,7 @@ def get_arguments(additional_args=[]):
     parser.add_argument('--N-X-FCP-seed', type=int, default=1010, help='seed used in train_helper')
     parser.add_argument('--N-X-FCP-h-dim', type=int, default=256, help='hidden dimension used in train_helper')
 
-    parser.add_argument('--exp-name', type=str, default='aamas25',
+    parser.add_argument('--exp-name', type=str, default='last',
                         help='Name of experiment. Used to tag save files.')
     parser.add_argument('--base-dir', type=str, default=Path.cwd(),
                         help='Base directory to save all models, data, wandbai.')
@@ -75,7 +75,7 @@ def get_arguments(additional_args=[]):
     parser.add_argument("--max-concurrent-jobs", type=int, default=None)
 
     parser.add_argument("--num-of-ckpoints", type=int, default=10)
-    parser.add_argument("--resume", action="store_true", default=False, help="Restart from last checkpoint")
+    parser.add_argument("--resume", action="store_true", default=False, help="Restart from last checkpoint for population training only")
 
     for parser_arg, parser_kwargs in additional_args:
         parser.add_argument(parser_arg, **parser_kwargs)

--- a/oai_agents/common/population.py
+++ b/oai_agents/common/population.py
@@ -62,7 +62,7 @@ def train_agent_with_checkpoints(args, total_training_timesteps, ck_rate, seed, 
     For SP agents, they only are trained with themselves so the order doesn't matter.
     '''
 
-    rlat.train_agents(total_train_timesteps=total_training_timesteps, tag=KeyCheckpoints.MOST_RECENT_TRAINED_MODEL, resume_ck_list=ck_rewards)
+    rlat.train_agents(total_train_timesteps=total_training_timesteps, tag_for_returning_agent=KeyCheckpoints.MOST_RECENT_TRAINED_MODEL, resume_ck_list=ck_rewards)
     checkpoints_list = rlat.ck_list
 
     if serialize:

--- a/scripts/train_agents.py
+++ b/scripts/train_agents.py
@@ -321,9 +321,9 @@ def set_input(args):
         args.n_envs = 2
         args.epoch_timesteps = 2
 
-        args.pop_total_training_timesteps = 3500 * 10
-        args.n_x_sp_total_training_timesteps = 1000 * 10
-        args.adversary_total_training_timesteps = 1000 * 10
+        args.pop_total_training_timesteps = 3500
+        args.n_x_sp_total_training_timesteps = 1000
+        args.adversary_total_training_timesteps = 1000
 
         args.fcp_total_training_timesteps = 1500
         args.n_x_fcp_total_training_timesteps = 1500 * 2
@@ -334,9 +334,8 @@ def set_input(args):
 
 if __name__ == '__main__':
     args = get_arguments()
-    args.quick_test = True
+    args.quick_test = False
     args.parallel = True
-    args.resume = True
 
     args.pop_force_training = False
     args.adversary_force_training = False
@@ -347,7 +346,7 @@ if __name__ == '__main__':
 
     set_input(args=args)
 
-    #SPN_1ADV_XSPCKP(args=args)
+    SPN_1ADV_XSPCKP(args=args)
 
     #SP(args)
 
@@ -359,4 +358,4 @@ if __name__ == '__main__':
 
     # SPN_XSPCKP(args=args)
 
-    N_1_FCP(args=args)
+    # N_1_FCP(args=args)

--- a/scripts/utils/train_helper.py
+++ b/scripts/utils/train_helper.py
@@ -34,7 +34,7 @@ def get_SP_agent(args, train_types, eval_types, curriculum, tag=KeyCheckpoints.M
         checkpoint_rate=args.pop_total_training_timesteps // args.num_of_ckpoints,
     )
 
-    selfplay_trainer.train_agents(total_train_timesteps=args.pop_total_training_timesteps, tag=tag)
+    selfplay_trainer.train_agents(total_train_timesteps=args.pop_total_training_timesteps, tag_for_returning_agent=tag)
     return selfplay_trainer.get_agents()[0]
 
 
@@ -169,7 +169,7 @@ def joint_ADV_N_X_SP(args, population, curriculum, unseen_teammates_len, adversa
                                             checkpoint_rate=total_train_timesteps // args.num_of_ckpoints,
                                             )
 
-        n_x_sp_types_trainer.train_agents(total_train_timesteps=total_train_timesteps, tag=tag)
+        n_x_sp_types_trainer.train_agents(total_train_timesteps=total_train_timesteps, tag_for_returning_agent=tag)
         agent_to_be_attacked = n_x_sp_types_trainer.get_agents()[0]
 
 
@@ -216,7 +216,7 @@ def no_ADV_N_X_SP(args, population, curriculum, unseen_teammates_len, n_x_sp_eva
                                         learner_type=args.primary_learner_type,
                                         checkpoint_rate=args.n_x_sp_total_training_timesteps // args.num_of_ckpoints,
                                         )
-    n_x_sp_types_trainer.train_agents(total_train_timesteps=args.n_x_sp_total_training_timesteps, tag=tag)
+    n_x_sp_types_trainer.train_agents(total_train_timesteps=args.n_x_sp_total_training_timesteps, tag_for_returning_agent=tag)
 
 
 
@@ -252,7 +252,7 @@ def get_adversary_agent(args, agent_to_be_attacked, attack_round, tag=KeyCheckpo
                                         hidden_dim=args.ADV_h_dim,
                                         learner_type=args.adversary_learner_type,
                                         checkpoint_rate=args.adversary_total_training_timesteps // args.num_of_ckpoints)
-    adversary_trainer.train_agents(total_train_timesteps=args.adversary_total_training_timesteps, tag=tag)
+    adversary_trainer.train_agents(total_train_timesteps=args.adversary_total_training_timesteps, tag_for_returning_agent=tag)
     return adversary_trainer.get_agents()[0]
 
 
@@ -305,7 +305,7 @@ def get_FCP_agent_w_pop(args,
         checkpoint_rate=args.fcp_total_training_timesteps // args.num_of_ckpoints,
     )
 
-    fcp_trainer.train_agents(total_train_timesteps=args.fcp_total_training_timesteps, tag=tag)
+    fcp_trainer.train_agents(total_train_timesteps=args.fcp_total_training_timesteps, tag_for_returning_agent=tag)
     return fcp_trainer.get_agents()[0], population
 
 
@@ -362,5 +362,5 @@ def get_N_X_FCP_agents(args,
         checkpoint_rate=args.n_x_fcp_total_training_timesteps // args.num_of_ckpoints,
     )
 
-    fcp_trainer.train_agents(total_train_timesteps=args.n_x_fcp_total_training_timesteps, tag=tag)
+    fcp_trainer.train_agents(total_train_timesteps=args.n_x_fcp_total_training_timesteps, tag_for_returning_agent=tag)
     return fcp_trainer.get_agents()[0], teammates_collection

